include/zephyr/drivers/accel2d.h

```diff
@@ -0,0 +1,680 @@
+ */
+int accel2d_draw_arc(const struct device *dev, struct accel_buf *buf,
+		     struct accel2d_src *src, struct accel2d_arc *arc,
+		     enum accel2d_blend blend);
+
+/**
+ * @brief Draw a line using the 2D accelerator
+ *
+ * Draws a line using the 2D accelerator hardware.
+ * @param dev: 2D accelerator device object
+ * @param buf: 2D accelerator output buffer description
+ * @param src: input source description
+ * @param start: start coordinates
+ * @param end: end coordinates
+ * @param width: width of line
+ * @param blend: 2D accelerator porter-duff blend setting
+ * @retval 0 command was queued successfully
+ * @retval -ENOSYS if this 2D accelerator does not support this feature
+ * @retval -ENOTSUP if the 2D accelerator does not support the
+ *          requested combination of settings
+ * @retval -EINVAL invalid buffer alignment settings for the 2D accelerator
+ * supporting this operation
+ * @retval -EBUSY @ref accel2d_render must be called before this operation can
+ *          proceed, as no more rendering can be queued on the 2D accelerator
+ * hardware.
+ * @retval -EIO Underlying 2D accelerator encountered an I/O error.
+ */
+int accel2d_draw_line(const struct device *dev, struct accel_buf *buf,
+		      struct accel2d_src *src, struct accel2d_coords *start,
+		      struct accel2d_coords *end, uint32_t width,
+		      enum accel2d_blend blend);
+
+/**
+ * @brief Draw an arbitrary gpu path
+ *
+ * This function leverages the 2D accelerator tesselation matrix and path opcode
+ * definitions to enable the user to draw an arbitrary shape using the
+ * 2D accelerator. The 2D accelerator path commands must form closed paths.
+ * Multiple paths can be drawn in one command by using the ACCEL2D_CLOSE_PATH
+ * opcode, followed by the ACCEL2D_MOVE opcode in the path definition to start a
+ * new path.
+ * @param dev: 2D accelerator device object
+ * @param buf: 2D accelerator output buffer description
+ * @param src: input source description
+ * @param path: uint32_t array of ACCEL2D_OP opcode definitions.
+ * @param path_len: length of path array, in bytes
+ * @param fill_rule: fill rule for path.
+ * @param transform: gpu affine transformation matrix.
+ * @param blend: 2D accelerator porter-duff blend setting
+ * @retval 0 command was queued successfully
+ * @retval -ENOSYS if this 2D accelerator does not support this feature
+ * @retval -ENOTSUP if the 2D accelerator does not support the
+ *          requested combination of settings
+ * @retval -EINVAL invalid buffer alignment settings for the 2D accelerator
+ * @retval -EBUSY @ref accel2d_render must be called before this operation can
+ *          proceed, as no more rendering can be queued on the 2D accelerator
+ * hardware.
+ * @retval -EIO Underlying 2D accelerator encountered an I/O error.
+ */
+int accel2d_draw_path(const struct device *dev, struct accel_buf *buf,
+		      struct gpu_src *src, uint32_t *path, uint32_t path_len,
```

Comment:
The `struct gpu_src` isn't available? Maybe some leftover?

```suggestion
		      struct accel2d_src *src, uint32_t *path, uint32_t path_len,
```